### PR TITLE
feat: add sum bench.

### PR DESCRIPTION
**Pull request by porcuquine**
This PR adds a simple benchmark that sums a vector of numbers. The numbers are provided as input, and for simplicity, the benchmark generates this as the vector of all numbers < N, when benching a vector of length N.

### CODE REVIEW

File: examples/sum/script/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+/// The ELF we want to execute inside the zkVM.
+const ELF: &[u8] = include_bytes!("../../program/elf/riscv32im-succinct-zkvm-elf");
+
+fn main() {
+    // Setup logging.
+    utils::setup_logger();
+
+    // Create an input stream and write input to it.
+    let nums = (0..1000).map(|i| i as u64).collect::<Vec<_>>();
```
Comment (huitseeker) [inline examples/sum/script/src/main.rs:11]: This is unidiomatic. I'd say `let nums: Vec<u64> = (0..1000).collect();`or `let nums = (0u64..1000).collect();`

File: examples/sum/.vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (huitseeker) [inline examples/sum/.vscode/settings.json:1]: This is a bit of a heavy config file in an example, could we remove it?

File: examples/sum/program/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+[workspace]
+[package]
+version = "0.1.0"
+name = "sum-program"
+edition = "2021"
+
+[dependencies]
+sp1-zkvm = { git = "https://github.com/succinctlabs/sp1.git", rev = "v1.0.0-testnet"  }
```
Comment (wwared) [inline examples/sum/program/Cargo.toml:8]: This should be
```suggestion
sphinx-zkvm = { path = "../../../zkvm/entrypoint" }
```
instead

File: examples/sum/.vscode/settings.json (hunk 2 above)
Comment (porcuquine) [reply examples/sum/.vscode/settings.json:1]: Sure, it was added by `cargo prove new`, fyi.

**Review by wwared — APPROVED** [review]
Nice!

**Pull request merged**